### PR TITLE
add dynamic filename

### PR DESCRIPTION
**Pull request by volbrene**
Idea #61

### CODE REVIEW

File: lib/interfaces/multer-extended-options.interface.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export interface MultipleSizeOptions {
 
 export interface MulterExtendedOptions extends Pick<MulterOptions, 'fileFilter' | 'limits'> {
   dynamicPath?: string;
+  dynamicFilename?: boolean;
```
Comment (jeffminsungkim) [inline lib/interfaces/multer-extended-options.interface.ts]: How about switching the variable name from `dynamicFilename` to `randomFilename`?
```suggestion
  randomFilename?: boolean;
```

File: lib/multer-sharp/interfaces/amazon-s3-upload-options.interface.ts [hunk 2]
```diff
@@ -4,4 +4,5 @@ export interface AmazonS3UploadOptions extends Partial<S3.Types.PutObjectRequest
   s3: S3;
   Key?: any;
   dynamicPath?: string;
+  dynamicFilename?: boolean;
```
Comment (jeffminsungkim) [inline lib/multer-sharp/interfaces/amazon-s3-upload-options.interface.ts]: Same here

File: lib/multer-sharp/multer-sharp.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ import {
   transformImage,
   isOriginalSuffix,
 } from './multer-sharp.utils';
+import { v4 as uuidv4 } from 'uuid';
```
Comment (jeffminsungkim) [inline lib/multer-sharp/multer-sharp.ts]: How about importing `randomStringGenerator` method from the Nest common utils instead?

```suggestion
import { randomStringGenerator } from '@nestjs/common/utils/random-string-generator.util';
```

Comment (jeffminsungkim) [reply lib/multer-sharp/multer-sharp.ts]: How about using template literals? They allow for both multiline strings and string interpolation.
```typescript
file.originalname =  `${randomStringGenerator()}.${extension(mimetype)}`
```

**Comment by volbrene** [comment]
@jeffminsungkim I have add some e2e tests for this problem. Hope that´s correct.

**Comment by volbrene** [comment]
@jeffminsungkim  done

**Comment by jeffminsungkim** [comment]
@volbrene 
Thank you very much!

@all-contributors please add @volbrene for code, test

**Comment by jeffminsungkim** [comment]
@volbrene 
I think the commit has an issue. Did you git commit with your correct Github account?

**Comment by volbrene** [comment]
Oh sorry, wrong account :) Can I change this?

**Comment by volbrene** [comment]
I have rename my author for the last commit. sorry for that, thats my first PR :)

**Comment by jeffminsungkim** [comment]
@volbrene 
Could you make a new PR for this?

**Comment by volbrene** [comment]
@jeffminsungkim New PR is there

**Pull request merged**